core/modules/syncer.js

```diff
@@ -358,189 +453,180 @@ Syncer.prototype.handleRefreshEvent = function() {
+				isReadyToSave = !tiddlerInfo || !tiddlerInfo.timestampLastSaved || tiddlerInfo.timestampLastSaved < thresholdLastSaved;
+			if(hasChanged) {
+				if(isReadyToSave) {
+					return new SaveTiddlerTask(this,title); 					
+				} else {
+					havePending = true;
 				}
-				// Mark that this task is no longer in progress
-				delete self.taskInProgress[task.title];
-				self.updateDirtyStatus();
-				// Process the next task
-				self.processTaskQueue.call(self);
-			});
-		} else {
-			// Make sure we've set a time if there wasn't a task to perform, but we've still got tasks in the queue
-			if(this.numTasksInQueue() > 0) {
-				this.triggerTimeout();
 			}
 		}
 	}
-};
-
-/*
-Choose the next applicable task
-*/
-Syncer.prototype.chooseNextTask = function() {
-	var self = this,
-		candidateTask = null,
-		now = Date.now();
-	// Select the best candidate task
-	$tw.utils.each(this.taskQueue,function(task,title) {
-		// Exclude the task if there's one of the same name in progress
-		if($tw.utils.hop(self.taskInProgress,title)) {
-			return;
-		}
-		// Exclude the task if it is a save and the tiddler has been modified recently, but not hit the fallback time
-		if(task.type === "save" && (now - task.lastModificationTime) < self.throttleInterval &&
-			(now - task.queueTime) < self.fallbackInterval) {
-			return;
-		}
-		// Exclude the task if it is newer than the current best candidate
-		if(candidateTask && candidateTask.queueTime < task.queueTime) {
-			return;
+	// Second, we check tiddlers that are known from the server but not currently in the store, and so need deleting on the server
+	titles = Object.keys(this.tiddlerInfo);
+	for(index=0; index<titles.length; index++) {
+		title = titles[index];
+		tiddlerInfo = this.tiddlerInfo[title];
+		tiddler = this.wiki.tiddlerExists(title) && this.wiki.getTiddler(title);
+		if(!tiddler) {
+			return new DeleteTiddlerTask(this,title);
 		}
-		// Now this is our best candidate
-		candidateTask = task;
-	});
-	return candidateTask;
+	}
+	// Check for tiddlers that need loading
+	title = Object.keys(this.titlesToBeLoaded)[0];
+	if(title) {
+		delete this.titlesToBeLoaded[title];
+		return new LoadTiddlerTask(this,title);
+	}
+	// No tasks are ready
+	return havePending;
 };
 
-/*
-Dispatch a task and invoke the callback
-*/
-Syncer.prototype.dispatchTask = function(task,callback) {
-	var self = this;
-	if(task.type === "save") {
-		var changeCount = this.wiki.getChangeCount(task.title),
-			tiddler = this.wiki.getTiddler(task.title);
-		this.logger.log("Dispatching 'save' task:",task.title);
-		if(tiddler) {
-			this.syncadaptor.saveTiddler(tiddler,function(err,adaptorInfo,revision) {
-				if(err) {
-					return callback(err);
-				}
-				// Adjust the info stored about this tiddler
-				self.tiddlerInfo[task.title] = {
-					changeCount: changeCount,
-					adaptorInfo: adaptorInfo,
-					revision: revision
-				};
-				// Invoke the callback
-				callback(null);
-			},{
```

Comment:
This is a breaking change/regression compared to a159b5baf3ad91d8defc68cbf81c78d01b69c416. This info should still be provided

Reply:
Hi @danielo515 sorry, what is it that has changed?

Reply:
Hello @Jermolene , take a look at the linked commit. This method was being called with an extra options object that included tiddlywiki information. I was sure I made a PR to include it back, but seems that I forgot to click the pull request button. Here it is:
https://github.com/Jermolene/TiddlyWiki5/pull/4584

Reply:
Ah, I see, thank you @danielo515